### ISSUE TITLE
Feature Request - make functions optional

### ISSUE THREAD

**Issue by Anto79-ops** [jekalmin/extended_openai_conversation#57]
Hey!

Would it be possible to make functions optional? For example, some models I use do not support functions and some partially do, and of course OpenAI does support it. 

you mentioned this modification to the code that I could use [here](https://github.com/jekalmin/extended_openai_conversation/issues/17#issuecomment-1829633611)  but then this would also break the OpenAI uses...which works really well, btw.

In the case where the model does not support functions, I would just like to use a the prompt and your integrations acts as a gateway to my localAI instance just acting as a conversation agent (or special prompt) without functions.

thanks

**Comment by jekalmin** [comment]
Thanks again for reporting an issue!

I will look into it in the next release.
I'm quite not sure how to reset functions to default (currently erasing all functions will set it to default).
However, making functions optional is needed as you mentioned.

**Comment by Anto79-ops** [comment]
thanks for being so responsive!

Yes, at the moment, blanking the functions text box (actually, I cannot leave it compleletely blank, I usually just put 1 space in the box), does work BUT it throws an core log error on every message it say:

```
Logger: custom_components.extended_openai_conversation
Source: custom_components/extended_openai_conversation/__init__.py:239
Integration: Extended OpenAI Conversation (documentation)
First occurred: November 26, 2023 at 7:58:41 PM (21 occurrences)
Last logged: 7:27:59 PM

Bad logger message: Failed to load functions ((TypeError("'NoneType' object is not subscriptable"),))
```

so if it can bypass the functions call when when functions are not needed, would be the best solution.

I'd be happy to try a beta if you wish.

Cheers

**Comment by jekalmin** [comment]
@Anto79-ops 
I released this in [0.0.10-beta2](https://github.com/jekalmin/extended_openai_conversation/releases/tag/0.0.10-beta2).

By setting `Functions` to `[]` like below, it will not pass `functions` parameter to openai
<img width="576" alt="스크린샷 2023-12-24 오후 10 49 41" src="https://github.com/jekalmin/extended_openai_conversation/assets/2917984/0f4dbae1-07a5-4153-87e3-739593c5d3cb">

Please try this and give a feedback :D
Thanks!

**Comment by Anto79-ops** [comment]
@jekalmin, thank you!

Just installed it and gave it a test, and it's working perfectly!

I can now use models that are not compatible with functions alongside OpenAI!

:)

**Comment by jekalmin** [comment]
Thanks for a quick repsonse!
I will let you know when it's released in stable version `0.0.10`.

**Comment by jekalmin** [comment]
Closing the issue.
Feel free to reopen it.
